frontend/catalyst/pennylane_extensions.py

```diff
@@ -122,46 +125,76 @@ def __init__(self, fn, device):
         self.device = device
         update_wrapper(self, fn)
 
+    @staticmethod
+    def _validate_qnode(qnode):
+        if not isinstance(qnode.device, qml.Device):
+            msg = "Device is not supported for compilation at the moment."
+            raise CompileError(msg)
+
+        name = qnode.device.short_name
+        is_known_device = name in QFunc.RUNTIME_DEVICES
+        implements_c_interface = hasattr(qnode.device, "get_c_interface")
+        is_valid_device = is_known_device or implements_c_interface
+        if not is_valid_device:
+            msg = f"The {name} device is not supported for compilation at the moment."
+            raise CompileError(msg)
+
+    @staticmethod
+    def _validate_library_path(lib_path):
+        if not pathlib.Path(lib_path).is_file():
+            msg = f"Device's library cannot be found at {lib_path}."
+            raise CompileError(msg)
+
+    @staticmethod
+    def get_library_path(device):
+        """Return C library for interfacing with the device.
+
+        For legacy purposes, returns the short name if no C library path is
+        implemented."""
+        if hasattr(device, "get_c_interface"):
+            lib_path = device.get_c_interface()
+            QFunc._validate_library_path(lib_path)
+            return lib_path
+
+        return device.short_name
+
+    @staticmethod
+    def _get_device_kwargs(qnode):
+        QFunc._validate_qnode(qnode)
+        name = qnode.device.short_name
+
+        # TODO:
+        # Once all devices get converted to shared libraries this name should just be the path.
+        device = qnode.device
+        backend_path = QFunc.get_library_path(device)
+
+        backend_kwargs = {}
+        if hasattr(device, "shots"):
+            backend_kwargs["shots"] = device.shots if device.shots else 0
+        if device.short_name == "braket.local.qubit":  # pragma: no cover
+            backend_kwargs["backend"] = device._device._delegate.DEVICE_ID
+        elif device.short_name == "braket.aws.qubit":  # pragma: no cover
+            backend_kwargs["device_arn"] = device._device._arn
+            if device._s3_folder:
+                backend_kwargs["s3_destination_folder"] = str(device._s3_folder)
+
+        # TODO: Maybe we should have a **device_kwargs stored in device.
+        # that way we avoid hand coding these and looking into the spec file
+        backend_kwargs["mcmc"] = getattr(device, "_mcmc", False)
+        backend_kwargs["num_burnin"] = getattr(device, "_num_burnin", 0)
+        backend_kwargs["kernel_name"] = getattr(device, "_kernel_name", None)
+        return {
+            "backend_name": name,
+            "backend_path": backend_path,
+            "backend_kwargs": backend_kwargs,
+        }
```

Comment:
@erick-xanadu please rebase this branch to `maa/split_devices_cmake` as there are some breaking changes related to the runtime methods in the frontend. This would reduce the amount of work required to update this PR after merging PR 343. Besides, we need this support to work with the plugin system :)

Reply:
Yeah, I was expecting this to happen. :)